hw/chip/mimxrt10xx/family.c

```diff
@@ -387,8 +387,28 @@ void USB_OTG1_IRQHandler(void)
 extern uint32_t _bootloader_dbl_tap;
 
 void board_check_app_start(void)
+
 {
-  uint32_t app_start_address = *(uint32_t *)(APP_START_ADDRESS + 4);
+  register uint32_t app_start_address = *(uint32_t *)(APP_START_ADDRESS + 4);
+
+  if (_bootloader_dbl_tap != DBL_TAP_MAGIC_QUICK_BOOT)
```

Comment:
Took me a moment to understand how we end up here, maybe add a comment about this condition

Reply:
Agreed, added.